de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/IHUAssignmentDAO.java

```diff
@@ -208,4 +210,14 @@
 	 * @return
 	 */
 	<T> List<T> retrieveModelsForHU(I_M_HU hu, Class<T> clazz, boolean topLevel);
+
+	/**
+	 * Retrieve the table hu assignments for the given HU even if they have LU and/or TU set. This is useful in the shipment hu assignments.
+	 * 
+	 * @param contextProvider
+	 * @param adTableId
+	 * @param hu
+	 * @return
+	 */
+	List<I_M_HU_Assignment> retrieveTableHUAssignmentsNoTopFilter(IContextAware contextProvider, int adTableId, I_M_HU hu);
```

Comment:
the adTableId parameter is ignored (note that this reported by codacy)
* isn't it required for the filtering? 
* pls either use it of remove it from the method signature